components/engine_traits/src/range_cache_engine.rs

```diff
@@ -41,6 +65,8 @@ pub trait RangeCacheEngine:
     // provide atomic write
     fn snapshot(
         &self,
+        region_id: u64,
+        region_epoch: u64,
```

Comment:
why not pass in &metapb::Region directly, so we have id, epoch and range

Reply:
Region's range is different from CacheRange(data range). I'm consider replace `Region` and `CacheRange` with：
```
struct CacheRegion {
  region_id: u64,
  epoch_version: u64,
  start: Vec<u8>,
  end: Vec<u8>,
}
```
Will try it in next PR.